### DISCUSSION TITLE
How to animate Visual.Bounds.Position?

### DISCUSSION THREAD

**Discussion by LITTOMA** [AvaloniaUI/Avalonia#14412] [Q&A] [answered]
I've tried below but got a `Unable to resolve type Bounds from namespace https://github.com/avaloniaui` error.

``` xaml
 <Grid.Transitions>
     <Transitions>
         <PointTransition Property="Bounds.Position" Duration="0:0:1" />
     </Transitions>
 </Grid.Transitions>
```

**Comment by maxkatz6** [comment]
You can try to implement `public class RectTransition : InterpolatingTransitionBase<Rect>` and use it on Bounds property, instead of PointTransition.

But for panel children transition you can also try Compositin API, which is more low level and harder to use, but will provide way better performance: [specifically ImplicitAnimations part](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/CompositionPage.axaml.cs#L97-L118).

**Reply by LITTOMA** [reply]
Thanks for your ideas. I tried to create a RectTransition and used it on Bounds property. But I got another error that says, "Cannot animate a direct property."

**Reply by maxkatz6** [reply]
Ah, yeah, you can't animate Bounds this way then.

**Reply by LITTOMA** [reply, marked as answer]
Never mind. I create KeyFrame animations using RenderTransfom and apply them on the control. They work for me.
